tensorflow/core/kernels/data/experimental/parallel_interleave_dataset_op.h

```diff
@@ -0,0 +1,61 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_CORE_KERNELS_DATA_EXPERIMENTAL_PARALLEL_INTERLEAVE_DATASET_OP_H_
+#define TENSORFLOW_CORE_KERNELS_DATA_EXPERIMENTAL_PARALLEL_INTERLEAVE_DATASET_OP_H_
+
+#include "tensorflow/core/framework/dataset.h"
+#include "tensorflow/core/kernels/data/captured_function.h"
+
+namespace tensorflow {
+namespace data {
+
+// See documentation in ../../ops/experimental_dataset_ops.cc for a high-level
+// description of the following op.
+
+class ParallelInterleaveDatasetOp : public UnaryDatasetOpKernel {
```

Comment:
Could we add a namespace `experimental`(`tensorflow::data::experimental::ParallelInterleaveDatasetOp`) to avoid the conflicts with another version of `ParallelInterleaveDatasetOp`(`tensorflow::data::ParallelInterleaveDatasetOp`)  defined at [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/data/parallel_interleave_dataset_op.h).

Reply:
Sure that sounds good and we should probably use it for all experimental ops.

Reply:
Yeah, will do that.